src/main/scala/is/hail/io/vcf/ExportVCF.scala

```diff
@@ -230,84 +166,106 @@ object ExportVCF {
         fatal(s"Invalid type for format field `${ fd.name }'. Found ${ fd.typ }.")
     }
   }
-
-  def apply[T >: Null](vkds: VariantSampleMatrix[Locus, Variant, T], path: String, append: Option[String] = None,
-    parallel: Boolean = false) {
-    val vas = vkds.vaSignature
-
-    val genotypeSignature = vkds.genotypeSignature
-
-    val (genotypeFormatField, genotypeFieldOrder) = genotypeSignature match {
-      case _: TGenotype =>
-        ("GT:AD:DP:GQ:PL", null)
-
-      case sig: TStruct =>
-        val fields = sig.fields
-        val formatFieldOrder: Array[Int] = sig.fieldIdx.get("GT") match {
-          case Some(i) => (i +: fields.filter(fd => fd.name != "GT").map(_.index)).toArray
-          case None => sig.fields.indices.toArray
-        }
-        val formatFieldString = formatFieldOrder.map(i => fields(i).name).mkString(":")
-
-        checkFormatSignature(sig)
-
-        (formatFieldString, formatFieldOrder)
-
-      case _ => fatal(s"Can only export to VCF with genotype signature of TGenotype or TStruct. Found `${ genotypeSignature }'.")
-    }
-
-    val infoSignature = vkds.vaSignature
-      .getAsOption[TStruct]("info")
-    val infoQuery: (Annotation) => Option[(Annotation, TStruct)] = infoSignature.map { struct =>
-      val (_, f) = vkds.queryVA("va.info")
-      (a: Annotation) => {
-        if (a == null)
-          None
-        else
-          Some((f(a), struct))
+  
+  def emitGenotype(sb: StringBuilder, formatFieldOrder: Array[Int], tg: TStruct, m: MemoryBuffer, offset: Long) {
+    formatFieldOrder.foreachBetween { j =>
+      val fIsDefined = tg.isFieldDefined(m, offset, j)
+      val fOffset = tg.loadField(m, offset, j)
+
+      tg.fields(j).typ match {
+        case TCall(_) =>
+          if (fIsDefined) {
+            strVCF(sb, TCall(), m, fOffset)
+          } else
+            sb.append("./.")
+        case it: TIterable =>
+          if (fIsDefined) {
+            val fLength = it.loadLength(m, fOffset)
+            iterableVCF(sb, it, m, fLength, fOffset)
+          } else
+            sb += '.'
+        case t =>
+          if (fIsDefined)
+            strVCF(sb, t, m, fOffset)
+          else
+            sb += '.'
```

Comment:
You can combine this with the `TCall` case and put the `TCall` test in the else branch.